File: mpp-proiect-repository-cristinahendre/lab9(Java)/Java/lab6/modul/RepoModule/src/main/java/Repository/DonatorRepo.java
package Repository;

import Domain.CazCaritabil;
import Domain.Donator;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;


import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class DonatorRepo implements  DonatorRepoI{

    private JdbcUtils utils;

    public DonatorRepo(Properties prop) {
        logger.info("intializare cu proprietati {}", prop);
        utils= new JdbcUtils(prop);
    }

    private  static  final Logger logger = LogManager.getLogger();


    @Override
    public Iterable<Donator> getDonatorDupaNume(String numeDat) {
        logger.traceEntry("afisez toti donatorii cu numele {} ", numeDat);
        List<Donator> lista = new ArrayList<>();
        Connection con = utils.getConnection();

        try(PreparedStatement pst = con.prepareStatement("select * from donator where nume like ? or prenume like ?")){
            pst.setString(1,numeDat+"%");
            pst.setString(2,numeDat+"%");

            try(ResultSet resultSet = pst.executeQuery()){
                while(resultSet.next()){
                    int id = resultSet.getInt("id");
                    String nume= resultSet.getString("nume");
                    String prenume =resultSet.getString("prenume");
                    String adresa= resultSet.getString("adresa");
                    long nr= resultSet.getLong("nr_telefon");

                    Donator donator = new Donator(nume,prenume,adresa,nr);
                    donator.setId(id);
                    lista.add(donator);
                }
            }
            //con.close();
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        logger.traceExit(lista);
        return  lista;

    }


    @Override
    public Donator getDonatorDupaDate(String nume, String prenume, String adresa, long telefon) {
        logger.traceEntry("caut donatorul cu aceste date: {},{},{},{} ", nume,prenume,adresa,telefon);

        Connection con = utils.getConnection();

        try(PreparedStatement pst = con.prepareStatement("select id from donator where nume =? and prenume= ? and adresa= ? and nr_telefon = ? ")){
            pst.setString(1,nume);
            pst.setString(2,prenume);
            pst.setString(3,adresa);
            pst.setLong(4,telefon);

            try(ResultSet resultSet = pst.executeQuery()){
                while(resultSet.next()){
                    int id = resultSet.getInt("id");


                    Donator donator = new Donator(nume,prenume,adresa,telefon);
                    donator.setId(id);
                    logger.traceEntry("Am gasit!!");
                    logger.traceExit(donator);
                    return donator;
                }
            }
           // con.close();
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }

        return  null;
    }

    @Override
    public Donator findOne(Integer integer) {
        logger.traceEntry("Caut donatorul cu acest id: {}", integer);

        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement("select * from donator where id = ?")){

            statement.setInt(1,integer);
            try(ResultSet resultSet = statement.executeQuery()){
                if(resultSet.next()){
                    int id = resultSet.getInt(1);
                    String nume= resultSet.getString(2);
                    String prenume= resultSet.getString(3);
                    String adresa= resultSet.getString(4);
                    long nrTelefon = resultSet.getLong(5);

                    Donator donator = new Donator(nume,prenume,adresa,nrTelefon);
                    donator.setId(id);
                    return donator;
                }
            }

            con.close();

        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return null;

    }

    @Override
    public Iterable<Donator> findAll() {
        logger.traceEntry("afisez toate elementele ");
        List<Donator> lista = new ArrayList<>();
        Connection con = utils.getConnection();

        try(PreparedStatement pst = con.prepareStatement("select * from donator")){

            try(ResultSet resultSet = pst.executeQuery()){
                while(resultSet.next()){
                    int id = resultSet.getInt("id");
                    String nume= resultSet.getString("nume");
                    String prenume =resultSet.getString("prenume");
                    String adresa= resultSet.getString("adresa");
                    long nr= resultSet.getLong("nr_telefon");

                    Donator donator = new Donator(nume,prenume,adresa,nr);
                    donator.setId(id);

                    lista.add(donator);
                }
            }
           // con.close();
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }

        return  lista;
    }

    @Override
    public Donator save(Donator entity) {
        logger.traceEntry("Adaug donatorul {}", entity);
        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement
                ("insert into donator(nume,prenume,adresa,nr_telefon) " +
                        "values (?,?,?,?)")){


            statement.setString(1,entity.getNume());
            statement.setString(2,entity.getPrenume());
            statement.setString(3,entity.getAdresa());
            statement.setLong(4,entity.getNrTelefon());

            int result= statement.executeUpdate();
            logger.traceEntry("S-au modificat {} randuri",result);
            con.close();
            return entity;



        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return  null;
    }

    @Override
    public void delete(Integer integer) {
        logger.traceEntry("Sterg donatorul cu id-ul {}", integer);
        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement
                ("delete from donator where id = ?")){

            statement.setInt(1,integer);

            int result= statement.executeUpdate();
            logger.traceEntry("S-au modificat {} randuri",result);
            con.close();




        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }


    }

    @Override
    public Donator update(Donator entity) {

        logger.traceEntry("Modific donatorul {}", entity);
        Connection con = utils.getConnection();

        try(PreparedStatement statement = con.prepareStatement
                ("update donator set nume=? , prenume =? ," +
                        "adresa =?,  nr_telefon =? where id =?")){

            statement.setString(1,entity.getNume());
            statement.setString(2,entity.getPrenume());
            statement.setString(3,entity.getAdresa());
            statement.setLong(4,entity.getNrTelefon());
            statement.setInt(5,entity.getId());

            int result= statement.executeUpdate();
            logger.traceEntry("S-au modificat {} randuri",result);
            con.close();
            return entity;



        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return  null;

    }

    @Override
    public int nrElem() {
        logger.traceEntry("Numar cate elemente sunt in donator");
        int nr =0;
        Connection con = utils.getConnection();
        try(Statement statement = con.createStatement()){

            ResultSet resultSet = statement.executeQuery("select count(*) from donator");
            resultSet.next();
            nr= resultSet.getInt(1);
            con.close();
            return nr;


        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        logger.traceExit(nr);
        return nr;
    }
}

File: mpp-proiect-repository-cristinahendre/C#/Cerinta1/Cerinta1/Repository/DonatorRepo.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Cerinta1.Domain;
using Cerinta1.Domain.Validator;
using log4net;

namespace Cerinta1.Repository
{
    class DonatorRepo : IDonatorRepo
    {

        private static readonly ILog log = LogManager.GetLogger("DonatorRepo");



        public Donator Delete(int id)
        {
            IDbConnection con = DBUtils.getConnection();
            log.InfoFormat("incerc sa sterg donatorul {0}", id);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from donator where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();
                con.Close();
                if (dataR == 0)
                {
                    log.InfoFormat("Eroare la stergerea donatorului.");
                    throw new RepositoryException("Nu s-a sters cu succes!");
                }
                else
                {
                    log.InfoFormat("Stergerea donatorului a avut loc cu succes.");
                    return null;
                }
            }
        }

        public IEnumerable<Donator> FindAll()
        {
            IList<Donator> lista = new List<Donator>();
            log.InfoFormat("Afisez toti donatorii");
            IDbConnection con = DBUtils.getConnection();
            try
            {

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select * from donator";


                    using (var dataR = comm.ExecuteReader())
                    {
                        while (dataR.Read())
                        {
                            int idV = dataR.GetInt32(0);
                            string nume = dataR.GetString(1);
                            string prenume = dataR.GetString(2);
                            string adresa = dataR.GetString(3);
                            long nr = dataR.GetInt64(4);

                            Donator don = new Donator(nume, prenume, adresa, nr);
                            don.Id = idV;

                            lista.Add(don);
                           
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            con.Close();
            log.InfoFormat("Am parcurs.");
            return lista;
        }



        public Donator FindOne(int id)
        {
            log.InfoFormat("Caut donatorul cu acest id: {0}", id);
            IDbConnection con = DBUtils.getConnection();
            try
            {

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select * from donator where id=@id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    comm.Parameters.Add(paramId);

                    using (var dataR = comm.ExecuteReader())
                    {
                        if (dataR.Read())
                        {
                            int idV = dataR.GetInt32(0);
                            string nume = dataR.GetString(1);
                            string prenume = dataR.GetString(2);
                            string adresa = dataR.GetString(3);
                            long nr = dataR.GetInt64(4);

                            Donator don = new Donator(nume, prenume, adresa, nr);
                            don.Id = idV;

                            log.InfoFormat("am gasit donatorul {0}", don);
                            con.Close();
                            return don;
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            log.InfoFormat("nu am  gasit acel donator.");
            return null;
        }



        public IEnumerable<Donator> getDonatorDupaNume(string nume)
        {
            IList<Donator> lista = new List<Donator>();
            log.InfoFormat("Afisez toti donatorii ce au numele/prenumele incepand cu {0}", nume);
            IDbConnection con = DBUtils.getConnection();
            try
            {

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select * from donator where nume like @nume or prenume like @pre";
                    IDbDataParameter paramNume = comm.CreateParameter();
                    paramNume.ParameterName = "@nume";
                    paramNume.Value = nume+"%";
                    comm.Parameters.Add(paramNume);


                    IDbDataParameter paramPrenume = comm.CreateParameter();
                    paramPrenume.ParameterName = "@pre";
                    paramPrenume.Value = nume + "%";
                    comm.Parameters.Add(paramPrenume);



                    using (var dataR = comm.ExecuteReader())
                    {
                        while (dataR.Read())
                        {
                            int idV = dataR.GetInt32(0);
                            string numeG = dataR.GetString(1);
                            string prenume = dataR.GetString(2);
                            string adresa = dataR.GetString(3);
                            long nr = dataR.GetInt64(4);

                            Donator don = new Donator(numeG, prenume, adresa, nr);
                            don.Id = idV;

                            lista.Add(don);

                        }
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            con.Close();
            log.InfoFormat("Am terminat.");
            return lista;
        }



        public int nrElem()
        {
            log.InfoFormat("Numar cati donatori sunt in baza de date.");
            IDbConnection con = DBUtils.getConnection();
            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select count(*) from donator";
                    int rez = int.Parse(comm.ExecuteScalar().ToString());
                    con.Close();
                    log.InfoFormat("Sunt {0} donatori.", rez);
                    return rez;
                }


            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            return -1;
        }



        public Donator Save(Donator entity)
        {
            log.InfoFormat("Voi salva donatorul {0}", entity);
            try
            {
                IDbConnection con = DBUtils.getConnection();
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "insert into donator(nume,prenume,adresa,nr_telefon)" +
                        " values (@nume,@pre,@addr,@tel)";

                   


                    IDbDataParameter paramNume = comm.CreateParameter();
                    paramNume.ParameterName = "@nume";
                    paramNume.Value = entity.Nume;
                    comm.Parameters.Add(paramNume);

                    IDbDataParameter paramPrenume = comm.CreateParameter();
                    paramPrenume.ParameterName = "@pre";
                    paramPrenume.Value = entity.Prenume;
                    comm.Parameters.Add(paramPrenume);

                    IDbDataParameter paramAdresa = comm.CreateParameter();
                    paramAdresa.ParameterName = "@addr";
                    paramAdresa.Value = entity.Adresa;
                    comm.Parameters.Add(paramAdresa);


                    IDbDataParameter paramTel = comm.CreateParameter();
                    paramTel.ParameterName = "@tel";
                    paramTel.Value = entity.NrTelefon;
                    comm.Parameters.Add(paramTel);

                    var dataR = comm.ExecuteNonQuery();
                    con.Close();
                    if (dataR == 0)
                    {
                        log.InfoFormat("Salvarea donatorului a intampinat probleme");
                        throw new RepositoryException("Nu s-a adaugat!");
                    }
                    else
                    {
                        log.InfoFormat("Salvarea donatorului e cu succes.");
                        return entity;
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }

            return null;
        }


        public Donator Update(Donator entity)
        {
            log.InfoFormat("Modific donatorul {0}", entity);
            try
            {
                IDbConnection con = DBUtils.getConnection();
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "update donator set nume=@nume, prenume=@pre, adresa=@addr," +
                        " nr_telefon =@tel where id = @id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = entity.Id;



                    IDbDataParameter paramNume = comm.CreateParameter();
                    paramNume.ParameterName = "@nume";
                    paramNume.Value = entity.Nume;
                    comm.Parameters.Add(paramNume);

                    IDbDataParameter paramPrenume = comm.CreateParameter();
                    paramPrenume.ParameterName = "@pre";
                    paramPrenume.Value = entity.Prenume;
                    comm.Parameters.Add(paramPrenume);

                    IDbDataParameter paramAdresa = comm.CreateParameter();
                    paramAdresa.ParameterName = "@addr";
                    paramAdresa.Value = entity.Adresa;
                    comm.Parameters.Add(paramAdresa);


                    IDbDataParameter paramTel = comm.CreateParameter();
                    paramTel.ParameterName = "@tel";
                    paramTel.Value = entity.NrTelefon;
                    comm.Parameters.Add(paramTel);


                    comm.Parameters.Add(paramId);

                    var dataR = comm.ExecuteNonQuery();
                    con.Close();
                    if (dataR == 0)
                    {
                        log.InfoFormat("Eroare la modificarea donatorului.");
                        throw new RepositoryException("Nu s-a modificat!");
                    }
                    else
                    {
                        log.InfoFormat("Donator modificat cu succes.");
                        return entity;
                    }
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            return null;
        }
    }
}
